tests/bayesian/experiment_suite.py

```diff
@@ -0,0 +1,226 @@
+                                                with open(filename, 'w') as outfile:
+                                                    json.dump(results, outfile, indent=4)
+
+                                            print "\n#################################"
+
+"""
+  Minimal version of the convergence_activity experiment from the htmpapers/frontiers repository.
+  Reproducing the experiment from the "How the neocortex learns the structure of the world" (Columns-paper).
+  Will be used to test and compare the results with the bayesian implementation of apical-TM and column pooler.
+"""
+
+def runExperiment(exp, numColumns):
+  """
+  We will run two experiments side by side, with either single column
+  or 3 columns
+  """
+  numFeatures = 3
+  numPoints = 10
+  numLocations = 10
+  numObjects = 10 # 2
+  numRptsPerSensation = 2
+
+  objectMachine = createObjectMachine(
+    machineType="simple",
+    numInputBits=20,
+    sensorInputSize=1024,
+    externalInputSize=1024,
+    numCorticalColumns=numColumns,
+    seed=40,
+  )
+  objectMachine.createRandomObjects(numObjects, numPoints=numPoints,
+                                    numLocations=numLocations,
+                                    numFeatures=numFeatures)
+
+  objects = objectMachine.provideObjectsToLearn()
+
+  # single-out the inputs to the column #1
+  objectsFeatures = {}
+  for i in range(numObjects):
+    featureLocations = []
+    for j in range(numLocations):
+      featureLocations.append({0: objects[i][j][0]})
+    objectsFeatures[i] = featureLocations
+
+  print "train "
+  exp.learnObjects(objectsFeatures)
+  # print "train multi-column "
+  # exp3.learnObjects(objects)
+
+  # test on the first object
+  objectId = 0
+  obj = objectMachine[objectId]
+
+  # Create sequence of sensations for this object for all columns
+  # We need to set the seed to get specific convergence points for the red
+  # rectangle in the graph.
+  objectSensations = {}
+  random.seed(12)
+  for c in range(numColumns):
+    objectCopy = [pair for pair in obj]
+    random.shuffle(objectCopy)
+    # stay multiple steps on each sensation
+    sensations = []
+    for pair in objectCopy:
+      for _ in xrange(numRptsPerSensation):
+        sensations.append(pair)
+    objectSensations[c] = sensations
+
+  sensationSteps = []
+  for step in xrange(len(objectSensations[0])):
+    pairs = [
+      objectSensations[col][step] for col in xrange(numColumns)
+    ]
+    sdrs = objectMachine._getSDRPairs(pairs)
+    sensationSteps.append(sdrs)
+
+  print "inference "
+  exp.sendReset()
+
+  # 1.) Inference: Get active predictions for each sensation/location input
+  l2ActiveCells = []
+  for sensation in sensationSteps:
+    exp.infer([sensation], objectName=objectId, reset=False)
+    rep = exp.getL2Prediction()
+    l2ActiveCells.append(rep)
+    activeCellNum = 0
+    for c in range(numColumns):
+        activeCellNum += len(rep[c])
```

Comment:
Why do we do that? Where do we use `activeCellNum` again?

Reply:
At the moment not, but left it in case we use multi-columns, we could check for total activity